mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala

```diff
@@ -70,9 +62,14 @@ class RankingMetrics[T: ClassTag] @Since("3.4.0") (
   @Since("1.2.0")
   def precisionAt(k: Int): Double = {
     require(k > 0, "ranking position k should be positive")
-    predictionAndLabels.map { case (pred, lab, _) =>
-      countRelevantItemRatio(pred, lab, k, k)
-    }.mean()
+    predictionAndLabels
```

Comment:
I think we may create a private rdd, and than use it internally to minimize changes.

something like this:
```
private val rdd = predictionAndLabels.map {
    case (pred: Array[T], lab: Array[T]) => ...
    case (pred: Array[T], lab: Array[T], rel: Array[Double]) => ...
}

```

Reply:
Do you mean that we use `rdd` as a private variable whose type is `RDD[(Array[T], Array[T], Array[Double])]`, and keep other methods almost the same ?